solutions/alt-tag-generator/pages/api/generate.ts

```diff
@@ -0,0 +1,51 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+type Data = string;
+interface ExtendedNextApiRequest extends NextApiRequest {
+  body: {
+    imageUrl: string;
+  };
+}
+
+export default async function handler(
+  req: ExtendedNextApiRequest,
+  res: NextApiResponse<Data>
+) {
+  const { imageUrl } = req.query;
+  // POST request to Replicate to start the alt generation process
+  let startResponse = await fetch("https://api.replicate.com/v1/predictions", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: "Token " + process.env.REPLICATE_API_KEY,
+    },
+    body: JSON.stringify({
+      version:
+        "2e1dddc8621f72155f24cf2e0adbde548458d3cab9f00c0139eea840d0ac4746",
+      input: { image: imageUrl },
+    }),
+  });
+
+  let jsonStartResponse = await startResponse.json();
+  let endpointUrl = jsonStartResponse.urls.get;
+
+  // GET request to get the status of the alt generation process & return the result when it's ready
+  let altText: string | null = null;
+  while (!altText) {
+    // Loop in 500ms intervals until the alt text is ready
+    let finalResponse = await fetch(endpointUrl, {
+      method: "GET",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: "Token " + process.env.REPLICATE_API_KEY,
+      },
+    });
+    let jsonFinalResponse = await finalResponse.json();
+    if (jsonFinalResponse.status === "succeeded") {
```

Comment:
Shouldn't check for a faild status and break? Or I was thinking about some maximum attempts, to not run this until termination.

Reply:
There is a type error after fail